### PR TITLE
:bug: Add back tolerations to cert-manager Deployments (backport to release-0.3)

### LINKED ISSUE #4276: cert-manager Tolerations removed
**What steps did you take and what happened:**
`clusterctl init` against a bootstrap cluster whose Nodes all have the `node-role.kubernetes.io/master:NoSchedule` Taint. The cert-manager Pods never got scheduled, because they are missing the Tolerations for that Taint.

**What did you expect to happen:**
I expected `init` to complete.

**Anything else you would like to add:**
#3034 added these Tolerations, and #3511 removed them, probably by accident.

Do we want to prevent this from happening again? One idea might be to add the Taint to the one-node kind clusters used by the e2e tests. Kind removes the Taint on such clusters by default. (Update: @fabriziopandini pointed to the right place to add a regression unit test, which is a more scoped way to verify this.)

**Environment:**

- Cluster-api version: master
- Minikube/KIND version: v0.10
- Kubernetes version: (use `kubectl version`): v1.20.4

/kind bug
/area clusterctl

### PR DESCRIPTION
**Pull request by dlipovetsky**
**What this PR does / why we need it**:
Backports #4277 to `release-0.3`

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #4276 

/milestone v0.3.x

### CODE REVIEW

**Comment by dlipovetsky** [comment]
I forgot to update a version constant:
```
 === RUN   Test_VersionMarkerUpToDate
--- FAIL: Test_VersionMarkerUpToDate (0.02s)
    cert_manager_test.go:52: 
        The cert-manager.yaml asset data has changed, but embeddedCertManagerManifestVersion and embeddedCertManagerManifestHash has not been updated.
        Expected
            <string>: "2a00ec..."
        to equal       |
            <string>: "fc399b..."
...
FAIL
FAIL	sigs.k8s.io/cluster-api/cmd/clusterctl/client/cluster	2.173s 
```

Fixed and force-pushed.

File: cmd/clusterctl/client/cluster/cert_manager.go [hunk 1]
```diff
@@ -52,7 +52,7 @@ const (
 	// NOTE: // If the cert-manager.yaml asset is modified, this line **MUST** be updated
 	// accordingly else future upgrades of the cert-manager component will not
 	// be possible, as there'll be no record of the version installed.
-	embeddedCertManagerManifestVersion = "v1.1.0"
+	embeddedCertManagerManifestVersion = "v1.1.1"
```
Comment (vincepri) [inline cmd/clusterctl/client/cluster/cert_manager.go:55]:
```suggestion
	embeddedCertManagerManifestVersion = "v1.1.0"
```

**Comment by fabriziopandini** [comment]
@dlipovetsky you have to update `embeddedCertManagerManifestHash` const after changing the generated bin data.
the `Test_VersionMarkerUpToDate` can be used to determine the new value.

**Comment by vincepri** [comment]
@dlipovetsky Are you able to fix the above?

File: cmd/clusterctl/client/cluster/cert_manager.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (dlipovetsky) [reply cmd/clusterctl/client/cluster/cert_manager.go]: Thanks for catching this. That version must have come from the master branch. Sorry, I missed your comment :disappointed:

**Comment by dlipovetsky** [comment]
Fixed version and force-pushed

**Comment by dlipovetsky** [comment]
Fixed the sha256 sum and force-pushed.

**Comment by dlipovetsky** [comment]
:sweat_smile: Just needs a /lgtm

**Pull request merged**